### ISSUE TITLE
Vertexs are not shared between faces

### ISSUE THREAD

**Issue by Vega-KH** [godotengine/godot#14469]
TL;DR - I THINK that Godot doesn't shared vertices between connected faces.  This seems inefficient.

When creating a mesh using ArrayMesh, you can share vertices. For example, a cube made up of 12 triangles, can be created using 8 vertices, with each of the corner vertices being shared by 3 triangles.

But each triangle requires 3 normals: one for each vertex of that triangle.  So in this case I would need 36 normals, while my vertex count is only 8.

However, using ArrayMesh, if the normal count doesn't match the vertex count, an error is thrown. It seems like sharing vertices isn't allowed in Godot.  (i.e. for a triangle mesh, the number of vertices must always be triangles x 3.) 

It would seem to me that this isn't at all efficient.  For a complex mesh, there are thousands of shared vertices.  But Godot is requiring that all of these be duplicated.  This equates to thousands of extra Vectors being stored in memory and passed back and forth to the graphics driver.  

Am I incorrect or missing something here?

**Comment by reduz** [comment]
You can't (mostly) share vertices in a cube, this is normal of any game engine due to how graphics APIs work.

**Comment by Vega-KH** [comment]
Ok, closing issue. I now realize it must be for good reason, though I don't understand it.  Thanks for feedback.
